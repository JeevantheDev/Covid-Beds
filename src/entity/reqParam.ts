/* eslint-disable @typescript-eslint/no-explicit-any */

export interface IcreateHospital {
  id?: number;
  userId?: number;
  HospitalBeds?: IcreateHospitalBeds[];
  createdAt?: Date | string;
  updatedAt?: Date | string;
  nameHospital?: string;
  locationType?: string;
  locationCoordinates?: string | string[];
  locationAddress?: string;
  locationFormattedAddress?: string;
  locationCity?: string;
  locationState?: string;
  locationZipcode?: string;
  locationCountryCode?: string;
  hospitalType?: string;
  hospitalEmail?: string;
  hospitalContactNo?: string;
  hospitalImage?: string;
}

export interface IcreateHospitalBeds {
  id?: number;
  hospitalName?: string;
  hospitalId?: number;
  totalBeds: number;
  currentBeds: number;
  createdAt?: Date | string;
  updatedAt?: Date | string;
}
