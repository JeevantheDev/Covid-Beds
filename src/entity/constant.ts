import { IcreateHospital } from './reqParam';

/* eslint-disable @typescript-eslint/no-explicit-any */
export interface IauthUser {
  email: string;
  name: string;
  authenticated: boolean;
}

export interface IcurrentUser {
  id?: string;
  name: string;
  email: string;
  emailVerified: string | null;
  image: string | null;
  Hospital: IcreateHospital[];
  createdAt: string;
  updatedAt: string;
}

export interface IqueryUser {
  email: string;
  name: string;
  image: string;
}

export interface Iprovider {
  callbackUrl: string;
  id: string;
  name: string;
  signinUrl: string;
  type: string;
}

export type ProviderTypes = {
  github?: Iprovider;
  google?: Iprovider;
  facebook?: Iprovider;
};

export enum HOSPITAL_TYPE {
  PVT = 'PVT',
  GOVT = 'GOVT',
}

export enum HOSPITAL_DEFAULT {
  POINT = 'Point',
  IMAGE = 'https://firebasestorage.googleapis.com/v0/b/blog-spot-e4923.appspot.com/o/images%2Fdownload.jfif?alt=media&token=86d1d01e-663e-4a64-b394-aa9a542163ad',
}
