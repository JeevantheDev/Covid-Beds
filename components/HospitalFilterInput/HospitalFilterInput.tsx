/* eslint-disable react/prop-types */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import React, { useState } from 'react';
import Grid from '@material-ui/core/Grid';
import { MenuItem, TextField, Button } from '@material-ui/core';
import { HOSPITAL_TYPE } from '../../src/entity/constant';

export interface Iprops {
  onFilterSubmit: (hospitalType: string) => void;
}

export const HospitalFilterInput: React.FC<Iprops> = ({ onFilterSubmit }) => {
  const [filterInput, setFilterInput] = useState<string>('any');

  const handleFilterInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    setFilterInput(e.target.value);
  };

  const handleFilterSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    onFilterSubmit(filterInput);
    setFilterInput('');
  };

  return (
    <form data-testid="filter-input-form">
      <Grid container spacing={2}>
        <Grid item xs={12} md={12} sm={12}>
          <TextField
            inputProps={{ 'data-testid': 'filterInput' }}
            onChange={handleFilterInput}
            value={filterInput}
            fullWidth
            select
            variant="outlined"
            name="filterInput"
          >
            <MenuItem value="any">Any</MenuItem>
            {[HOSPITAL_TYPE.GOVT, HOSPITAL_TYPE.PVT].map((option) => (
              <MenuItem key={option} value={option}>
                {option}
              </MenuItem>
            ))}
          </TextField>
        </Grid>
        <Grid item xs={12} md={12} sm={12}>
          <Button
            data-testid="submit"
            onClick={handleFilterSubmit}
            style={{ width: '100%' }}
            color="secondary"
            variant="contained"
          >
            Find Hospital
          </Button>
        </Grid>
      </Grid>
    </form>
  );
};
