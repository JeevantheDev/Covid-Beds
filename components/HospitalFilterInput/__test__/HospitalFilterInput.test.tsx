/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { HospitalFilterInput, Iprops } from '../HospitalFilterInput';
import '@testing-library/jest-dom/extend-expect';
import TestRenderer from 'react-test-renderer';
import { HOSPITAL_TYPE } from '../../../src/entity/constant';

afterEach(cleanup);

function renderFilterInputForm(props: Partial<Iprops> = {}) {
  const defaultProps: Iprops = {
    onFilterSubmit(hospitalType: string) {
      return;
    },
  };

  return render(<HospitalFilterInput {...defaultProps} {...props} />);
}

test('should display a blank filter input', async () => {
  const { findByTestId } = renderFilterInputForm();
  const filterInputForm = await findByTestId('filter-input-form');
  expect(filterInputForm).toHaveFormValues({ filterInput: 'any' });
});

test('should allow user to submit for filter', async () => {
  const onFilterSubmit = jest.fn();
  const { findByTestId } = renderFilterInputForm({ onFilterSubmit });
  const filterInput = await findByTestId('filterInput');
  const submit = await findByTestId('submit');

  fireEvent.change(filterInput, { target: { value: HOSPITAL_TYPE.PVT } });
  fireEvent.click(submit);

  expect(onFilterSubmit).toHaveBeenCalledWith('PVT');
});

it('matches snapshot', () => {
  const onFilterSubmit = jest.fn();
  const tree = TestRenderer.create(
    <HospitalFilterInput onFilterSubmit={onFilterSubmit}></HospitalFilterInput>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
