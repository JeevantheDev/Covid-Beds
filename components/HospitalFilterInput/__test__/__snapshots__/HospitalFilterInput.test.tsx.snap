// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  data-testid="filter-input-form"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-component-select-filterInput"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="mui-component-select-filterInput"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Any
          </div>
          <input
            aria-hidden={true}
            autoFocus={false}
            className="MuiSelect-nativeInput"
            data-testid="filterInput"
            name="filterInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="any"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-10"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        data-testid="submit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Find Hospital
        </span>
      </button>
    </div>
  </div>
</form>
`;
